Show all Orgs that a user is a member of in user's dashboard

Waterdrips opened the pull request:
## Description
We can now display all organisation's functions when a user is a public member and Oauth is enabled.
This shows as Org functions in a new column on dashboard with links pointing to that organisation's urls (a user will then be in that org's dashboard area, the breadcrumb link will be back to that org)

Lots of this code is from an old PR,
closes #347 
closes #361 

See screenshot:
<img width="1204" alt="Screenshot 2019-12-04 at 08 10 28" src="https://user-images.githubusercontent.com/40488132/70124856-96794e00-166d-11ea-8d14-ef95aa804d7b.png">





## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->
Deployed to my own OFC installation, all links checked from dashboard etc.

## How are existing users impacted? What migration steps/scripts do we need?
New deployment of the system-dashboard.
This is extra functionality and should improve usability of the OFC dashboard.


## Checklist:

I have:

- [ ] updated the documentation and/or roadmap (if required)
- [x] read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] signed-off my commits with `git commit -s`
- [ ] added unit tests

alexellis commented:
Thanks for working on this. Can I try it out somewhere?

Waterdrips commented:
@alexellis is authorised on my OFC installation: https://system.heyal.uk/dashboard/

I have just deployed from the tip of this branch, so it matches. 

If you want to install the dashboard on your own cluster you can use this image: 
`heyal/of-cloud-dashboard:demo-v1`

alexellis commented:
I'm seeing OF which also stands for OpenFaaS. This is a little confusing. I wonder if we need to rethink the original idea about shortening to initials? How is the colour calculated? On full name or initials? What if you add an org called OpenFields or similar?

Waterdrips commented:
I did look at grabbing GitHub avatars, but this wouldn't work for gitlab. 

In terms of colour and initials this is code that was pulled from the old PR (#361)
Colour functions point at this : https://stackoverflow.com/questions/3426404/create-a-hexadecimal-colour-based-on-a-string-with-javascript

Initials come from 
```js
const genOwnerInitials = (owner) => (
  (owner.split(/[-_]+/).length > 1) ?
    owner.split(/[-_]+/)[0].substring(0, 1).concat(owner.split(/[-_]+/)[1].substring(0, 1)) :
      owner.split(/[-_]+/)[0].substring(0, 2)

);
```
If the owner has a - or _, use the first letter of first 2 sections
If not, use first 2 letters

Might be better without this stuff, and just use the org/user name

Waterdrips commented:
hi @alexellis, iv rebased this pr and re-tested (With and without auth)

Have you thought any more about the initials vs full org/username in the table?

alexellis commented:
> Might be better without this stuff, and just use the org/user name

^ In the screenshot it would probably take the same amount of room, let's err on the side of simple and try that?

Waterdrips commented:
@alexellis - This is without formatting
<img width="1241" alt="Screenshot 2019-12-02 at 16 39 02" src="https://user-images.githubusercontent.com/40488132/69977311-7db05180-1522-11ea-9fde-53531d3c3ac9.png">

alexellis commented:
Can I try this? Do you have an environment up?

alexellis commented:
How does it look on mobile now for responsive web? Can you take a screenshot in the Chrome preview?

Waterdrips commented:
> @alexellis is authorised on my OFC installation: https://system.heyal.uk/dashboard/
> 
> I have just deployed from the tip of this branch, so it matches.

^ Its available there.


Responsive view of the table is not ideal, it scrolls sideways, this has not changed from previous versions.
<img width="270" alt="Screenshot 2019-12-06 at 09 24 50" src="https://user-images.githubusercontent.com/40488132/70311919-759a2f80-180a-11ea-91ba-c706ad55d5f3.png">
<img width="259" alt="Screenshot 2019-12-06 at 09 24 56" src="https://user-images.githubusercontent.com/40488132/70311920-759a2f80-180a-11ea-8661-f3df82030051.png">

alexellis commented:
What do we want to drop for this viewport?

alexellis commented:
May I suggest: invocations replicas sha?

Waterdrips commented:
How about this: 
<img width="251" alt="Screenshot 2019-12-07 at 11 24 42" src="https://user-images.githubusercontent.com/40488132/70373855-2cb6a980-18e4-11ea-99b3-92114229e2a9.png">

alexellis commented on dashboard/client/src/components/FunctionTableItem/FunctionTableItem.jsx:
```diff
@@ -20,7 +20,14 @@ const genRepoUrl = ({ gitRepoURL, gitBranch }) => {
 };
 
-const FunctionTableItem = ({ onClick, fn, user }) => {
+const genOwnerInitials = (owner) => (
```
Do we still need this function?

Waterdrips replied:
no

alexellis commented on dashboard/of-cloud-dashboard/handler.js:
```diff
@@ -207,7 +212,9 @@ var isResourceInTokenClaims = function (path, queryString, user, organisations)
   }
-  return organisations.indexOf(queryString["user"]) >= 0
+
+  let orgs = organisations.split(",");
```
Can `organisations` be undefined? What if it was zero lengthed?

Waterdrips replied:
No it can't be undef, if we dont have a cookie we return above, otherwise its an empty string.

and "".split(",") --> [""], so no error

Waterdrips replied:
No it can't be undef, if we dont have a cookie we return above, otherwise its an empty string.

and "".split(",") --> [""], so no error
